File: internal/ports/outbound/repository/session.ts
import type { Session } from '@/internal/domain/model/session';
import type { PaginationMeta } from '@/internal/domain/model/error';

export interface ISessionRepository {
  list(params?: { page?: number; limit?: number }): Promise<{ data: Session[]; pagination: PaginationMeta }>;
  getById(id: number): Promise<Session>;
  create(userId: number, nodeId: number): Promise<{ id: number }>;
  end(payload: { id?: number; user_id?: number; node_id?: number }): Promise<void>;
  delete(id: number): Promise<void>;
}

File: backend/internal/ports/inbound/http/session.go
package portsinboundhttp

import (
	"context"

	domainmodel "github.com/Bengkel-Inovasi/GeLoRa/backend/internal/domain/model"
)

type Session interface {
	AddSession(ctx context.Context, userId int64, nodeId int64) (id int64, err error)
	GetSessionById(ctx context.Context, id int64) (session *domainmodel.Session, err error)
	GetSessionsList(ctx context.Context, page int, limit int, cursorId *int64, userId *int64, nodeId *int64, active *bool) (sessions []domainmodel.Session, total int, err error)
	SetSessionEndSession(ctx context.Context, id *int64, userId *int64, nodeId *int64) (err error)
	RemoveSessionById(ctx context.Context, id int64) (err error)
}
